### ISSUE TITLE
bug: Long documents are stuck in RAG

### ISSUE THREAD

**Issue by 0xSage** [janhq/jan#2795]
**Describe the bug**

Cannot RAG with super long PDFs.

e.g. https://arxiv.org/pdf/2404.14619.pdf

**Steps to reproduce**
1. Upload long PDF
2. Model starts
3. Generation never completes
4. No error msgs

**Expected behavior**
An error message: i.e. some user friendly version of "
2024-04-24T05:33:22.626Z [NITRO]::Error: llama_get_embeddings_ith: invalid embeddings id 817, reason: out of range [0, 512)"

**Screenshots**
If applicable, add screenshots to help explain your issue.

**Environment details**
- Operating System: [Specify your OS. e.g., MacOS Sonoma 14.2.1, Windows 11, Ubuntu 22, etc]
- Jan Version: [e.g., 0.4.3]
- Processor: [e.g., Apple M1, Intel Core i7, AMD Ryzen 5, etc]
- RAM: [e.g., 8GB, 16GB]
- Any additional relevant hardware specifics: [e.g., Graphics card, SSD/HDD]

**Logs**
```sh

20240424 05:33:05.305918 UTC 14668889 ERROR failed to get embeddings token 28770, seq_id 0 - llama_server_context.h:1275

2024-04-24T05:33:09.771Z [NITRO]::Error: llama_get_embeddings_ith: invalid embeddings id 1009, reason: out of range [0, 512)

2024-04-24T05:33:09.771Z [NITRO]::Debug: 20240424 05:33:05.306267 UTC 14668889 DEBUG [update_slots] slot 0 released (1678 tokens in cache) - llama_server_context.h:1596
20240424 05:33:05.306277 UTC 14668889 DEBUG [update_slots] all slots are idle and system prompt is empty, clear the KV cache - llama_server_context.h:1540
20240424 05:33:05.307174 UTC 14668889 DEBUG [launch_slot_with_data] slot 0 is processing [task id: 99] - llama_server_context.h:876
20240424 05:33:05.309381 UTC 14668889 DEBUG [update_slots] slot 0 : kv cache rm - [0, end) - llama_server_context.h:1735
20240424 05:33:09.770970 UTC 14668889 ERROR failed to get embeddings token 28783, seq_id 0 - llama_server_context.h:1275

2024-04-24T05:33:16.969Z [NITRO]::Error: llama_get_embeddings_ith: invalid embeddings id 1625, reason: out of range [0, 512)

2024-04-24T05:33:16.969Z [NITRO]::Debug: 20240424 05:33:09.771209 UTC 14668889 DEBUG [update_slots] slot 0 released (1010 tokens in cache) - llama_server_context.h:1596
20240424 05:33:09.771222 UTC 14668889 DEBUG [update_slots] all slots are idle and system prompt is empty, clear the KV cache - llama_server_context.h:1540
20240424 05:33:09.772188 UTC 14668889 DEBUG [launch_slot_with_data] slot 0 is processing [task id: 100] - llama_server_context.h:876
20240424 05:33:09.775826 UTC 14668889 DEBUG [update_slots] slot 0 : kv cache rm - [0, end) - llama_server_context.h:1735
20240424 05:33:16.968896 UTC 14668889 ERROR failed to get embeddings token 574, seq_id 0 - llama_server_context.h:1275

2024-04-24T05:33:22.626Z [NITRO]::Error: llama_get_embeddings_ith: invalid embeddings id 817, reason: out of range [0, 512)

2024-04-24T05:33:22.626Z [NITRO]::Debug: 20240424 05:33:16.969120 UTC 14668889 DEBUG [update_slots] slot 0 released (1626 tokens in cache) - llama_server_context.h:1596
20240424 05:33:16.969128 UTC 14668889 DEBUG [update_slots] all slots are idle and system prompt is empty, clear the KV cache - llama_server_context.h:1540
20240424 05:33:16.970503 UTC 14668889 DEBUG [launch_slot_with_data] slot 0 is processing [task id: 101] - llama_server_context.h:876
20240424 05:33:16.971972 UTC 14668889 DEBUG [update_slots] slot 0 : kv cache rm - [0, end) - llama_server_context.h:1735
20240424 05:33:19.029754 UTC 14668889 DEBUG [update_slots] slot 0 released (462 tokens in cache) - llama_server_context.h:1596
20240424 05:33:19.029773 UTC 14668889 DEBUG [update_slots] all slots are idle and system prompt is empty, clear the KV cache - llama_server_context.h:1540
20240424 05:33:19.030659 UTC 14668889 DEBUG [launch_slot_with_data] slot 0 is processing [task id: 102] - llama_server_context.h:876
20240424 05:33:19.031862 UTC 14668889 DEBUG [update_slots] slot 0 : kv cache rm - [0, end) - llama_server_context.h:1735
20240424 05:33:22.625860 UTC 14668889 ERROR failed to get embeddings token 28734, seq_id 0 - llama_server_context.h:1275


```

**Comment by hahuyhoang411** [comment]
The team is working on refactoring RAG for the best experience. Please stay tuned.

**Comment by kalle07** [comment]
feat: doclibrary for multible pdf #2852 
and yes, long pdf dont work
(copy and paste it from GPT4all)  ;)
also usefull to choose my own embedding model like nomic or other

**Comment by imtuyethan** [comment]
Should be solved once we reimplement #3026?

**Comment by imtuyethan** [comment]
- Closed as a sub-issue of https://github.com/janhq/cortex-internal/issues/20
